### PR TITLE
[SPARK-31150][SQL][Followup] Handle ' as escape for text

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

pattern `''` means literal `'` 

```sql
select date_format(to_timestamp("11111904-01-23 15:02:01", 'y-MM-dd HH:mm:ss'), "y-MM-dd HH:mm:ss''SSSSSSSSS");
5377-02-14 06:27:19'000000519
```
https://github.com/apache/spark/commit/0946a9514f56565c78b0555383c1ece14aaf2b7b missed this case and this pr add it back.

### Why are the changes needed?

bugfix

### Does this PR introduce any user-facing change?

no
### How was this patch tested?

add ut

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala [hunk 1]
```diff
@@ -117,6 +117,7 @@ private object DateTimeFormatterHelper {
       pattern: String): DateTimeFormatterBuilder = {
     val builder = createBuilder()
     pattern.split("'").zipWithIndex.foreach {
+      case ("", idx) if idx != 0 => builder.appendLiteral("'")
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala]: can we add some comment to explain why `idx != 0`?

Comment (yaooqinn) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala]: comment add, thanks.

**Review by cloud-fan — APPROVED** [review]

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
Thanks! Merged to master/3.0.

**Comment by gatorsmile** [comment]
@yaooqinn Please fix the title next time.

**Comment by yaooqinn** [comment]
thanks

**Pull request closed without merging**